core/src/main/java/org/infinispan/marshall/core/ClassIdentifiers.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.marshall.core;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.container.entries.InternalCacheValue;
+
+public class ClassIdentifiers implements InternalClassIds {
+   private final static ClassIdentifiers INSTANCE = new ClassIdentifiers();
```

Comment:
This is very dangerous. You have a static reference to a class that contains a reference to a collection whose values are classes. You're essentially creating a classloader leak in a managed environment.

Given the dangers, I'd avoid the singleton static class here. 

Also, if you instantiate it in GlobalMarshaller, make sure you clear it when it stops.